src/runtime-rs/crates/resource/src/share_fs/share_uds.rs

```diff
@@ -0,0 +1,144 @@
+                                                if let Ok(mut hside) = UnixStream::connect(&volume).await {
+                                                    let nvolume = volume.clone();
+                                                    tokio::spawn(async move {
+                                                        debug!(sl!(), "do iocopy for share_uds:{} between guest and host", &nvolume);
+                                                        let res = tokio::io::copy_bidirectional(&mut stream, &mut hside).await;
+                                                        debug!(sl!(), "io copy terminated for {} with {:?} between guest and host", &nvolume, res);
+                                                    });
+                                                }
+                                            }
+                                        },
+                                        _ = notifier.notified() => {
+                                            info!(sl!(), "destroy the share_uds {}", &volume);
+                                            break
+                                        }
+                                    }
+                                }
+                                Listener::Vsock(ref mut listener) => {
+                                    tokio::select! {
+                                        new_conn = listener.accept() => {
+                                            if let Ok((mut stream, _addr)) = new_conn {
+                                                debug!(sl!(), "share_uds {} got new vsock connection from guest", &volume);
+                                                    if let Ok(mut hside) = UnixStream::connect(&volume).await {
+                                                        let nvolume = volume.clone();
+                                                        tokio::spawn(async move {
+                                                            debug!(sl!(), "do iocopy for share_uds:{} between guest and host", &nvolume);
+                                                            let res = tokio::io::copy_bidirectional(&mut stream, &mut hside).await;
+                                                            debug!(sl!(), "io copy terminated for {} with {:?} between guest and host", &nvolume, res);
+                                                        });
+                                                    }
+
+
+                                            }
+                                        },
+                                        _ = notifier.notified() => {
+                                            info!(sl!(), "destroy the share_uds {}", &volume);
+                                            break
+                                        }
+                                    }
+                                }
```

Comment:
A broader point: `Listener::Unix` and `Listener::Vsock` handling look pretty much identical (the only difference I was able to notice is in a debug log message wording).  It's a significant amount of non-trivial code, can we only have it once in the codebase?  If adding a named function doesn't seem justified, even just a lambda right before the `match` statement could help a lot, what do you think?

Reply:
Hi  @pmores 

In fact, the difference here is the difference between VsockListener and UnixListener. It is difficult to merge these two different types into one function to process. Of course, we can define a trait for these two types outside. However, the streams returned by these two types of listen are two different types: UnixStream and VsockStream. We need to define a trait for these two. Doing so seems to combine the two processes, but it introduces the definition of two traits, and the result does not make the code more concise.